doc/decisions/1_in_discussion/spec_expressiveness.md

```diff
@@ -2,22 +2,155 @@
 
 ## Problem
 
-Currently, you can easily come into wrong assumptions that something would work in the specification.
+Specifications can help users define certain details about keys and make sure only valid keys are used.
 
-We need to find minimal requirements to implement a sane spec plugin.
+One example of such as specification can be found below:
+
+```ni
+[public/ip]
+meta:/default = 127.0.0.1
+meta:/description = "The IP via which this application is accessible."
+
+[public/port]
+meta:/default = 8080
+meta:/description = "The port on which this application is accessible."
+```
+
+Here one can see `default` and `description` used as metadata. 
+
+The current implementation of the spec plugin is copying this metadata to all other namespaces.
+It also assures that if keys do not exist and have `default` metadata, it creates a cascading key.
+
+Assume we have a REST application we want to write a specification for.
+This REST application has to use multiple services.
+
+If we have `database`, `broker` and `mailer` services we could use the specification:
+
+```ni
+[service/database/port]
+meta:/require = true
+meta:/default = 8080
+meta:/description = "The port of the database service."
+
+[service/database/url]
+meta:/require = true
+meta:/description = "The url of the database service."
+
+[service/broker/port]
+meta:/require = true
+meta:/default = 8080
+meta:/description = "The port of the broker service."
+
+[service/broker/url]
+meta:/require = true
+meta:/description = "The url of the broker service."
+
+[service/mailer/port]
+meta:/require = true
+meta:/default = 8080
+meta:/description = "The port of the mailer service."
+
+[service/mailer/url]
+meta:/require = true
+meta:/description = "The url of the mailer service."
+```
+
+This is already very repetitive and error prone. 
+But writing the specification becomes completely impossible when the set of services is not known in advance. 
+To solve this we need some way to allow for dynamic/unknown parts in the keyname.
+
+`Globbing` expressions is the solution to this problem. 
```

Comment:
The problem part should only contain the problem.

Reply:
I already have just written down the problems in a previous version. I was told to back it up with some example, which was a very good input IMO, as I think people reading this document immediately understand what this problem is about without the need of knowing to much details.